### PR TITLE
Add ghc-8.0.1 to travis matrix

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

**Comment by phadej** [comment]
This is not easily fixable as `should-not-typecheck` hack doesn't work with `ghc-8.0.1` : https://github.com/CRogers/should-not-typecheck/pull/6

**Comment by kosmikus** [comment]
@phadej I think it's fixable. Commented there.

**Comment by phadej** [comment]
next issue is `testing-feat` dependency. There is no repository/issues link in http://hackage.haskell.org/package/testing-feat. Should we send an email to the maintainer?

**Comment by utdemir** [comment]
@phadej I had found a GitHub repo of `testing-feat` and opened an issue on 28th January, but sadly there isn't any answer yet:

https://github.com/JonasDuregard/testing-feat/issues/3

**Comment by phadej** [comment]
@utdemir I made a fix, hopefully maintainer could accept it and release `0.4.0.3` soon-ish. 

Otherwise, if servant-maintainers think this is blocking issue, they can request NMU from hackage-trustees.

**Comment by rvion** [comment]
[testing-feat-0.4.0.3](http://hackage.haskell.org/package/testing-feat-0.4.0.3) has been released

**Comment by soenkehahn** [comment]
I've restarted the travis build.

**Comment by phadej** [comment]
Now failing because of https://github.com/ekmett/lens/issues/649

**Comment by phadej** [comment]
Finally got this green \o/

File: .travis.yml [hunk 1]
```diff
@@ -3,22 +3,25 @@ sudo: false
 language: c
 
 env:
-  - GHCVER=7.8.4
-  - GHCVER=7.10.2
+  - GHCVER=7.8.4 CABALVER=1.22
```
Comment (jkarni) [inline .travis.yml]: Does 1.24 not work with 7.8.4?

**Comment by jkarni** [comment]
LGTM

File: .travis.yml (hunk 1 above)
Comment (phadej) [inline .travis.yml]: It works. I can change it to use `1.24`.

Actually then one could change to use [`new-build`](http://blog.ezyang.com/2016/05/announcing-cabal-new-build-nix-style-local-builds/) which would make `tinc` obsolete. But maybe not too much changes at once

**Comment by alpmestan** [comment]
@phadej Cheers! Great news. It'll be nice when there's an LTS or something for ghc 8, that'll cut down significantly the length of the corresponding stack YAML file =)

In the meantime, LGTM.

File: .travis.yml (hunk 1 above)
Comment (jkarni) [inline .travis.yml]: Let's leave discussing `tinc` vs `new-build` for a different PR then. 

I'm okay either with 1.22 or 1.24 - on the one hand, having different ones maybe helps check that everything works with both; on the other, it's a little cleaner with a single one.

**Comment by jkarni** [comment]
Thanks a lot! I added changelog updates as well.

**Pull request merged**